modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -499,6 +502,120 @@ class NaryEltwiseLayerImpl CV_FINAL : public NaryEltwiseLayer
                 f, scale, ninputs, max_ndims, shapes[0], inp, out, (const size_t **) steps, ptrs);
     }
 
+    template <typename T, typename Functor>
+    void trinary_forward(const Functor& f, const std::vector<Mat>& inputs, std::vector<Mat>& outputs)
```

Comment:
To be honest it looks like we could change the original function to accept multiple input vector.
E.g. make it accept (data + step) buffer. And everywhere where there is $i suffix (e.g. ptr1_, etc), we could use buffer as well.
I think we had performance tests somewhere, we should check that it didn't degrade. If it does, we could have 1) fast binary path and 2) slower naray path.

Reply:
Hi, @rogday. Thanks for the code review.
I'm not very familiar with this part of the code, can you give some examples in more detail so I can complete it?